### PR TITLE
feat: Support multi urls matching.

### LINKED ISSUE #124: Feature: Enhance hook matching.
### Checklist

- [X] I've searched for similar issues and couldn't find anything matching
- [X] I've discussed this feature request in the [OpenIMSDK Slack](https://join.slack.com/t/openimsdk/shared_invite/zt-1tmoj26uf-_FDy3dowVHBiGvLk9e5Xkg) and got positive feedback

### Is this feature request related to a problem?

❎ No

### Problem Description

Support hook with multi url matching.

```go
func (h CROS) Pattern() string {
	return "/*"
}
````
To
```go
func (h CROS) Patterns() []string {
	return []string{
		"/aaa",
		"/bbb",
	}
}
```

### Solution Description

Update url string to url string array.

### Benefits

Enhance system.

### Potential Drawbacks

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by IRONICBo**
#### 🔍 What type of PR is this?

/kind feature

#### 👀 What this PR does / why we need it:

Support multi url hook matching.

#### 🅰 Which issue(s) this PR fixes:

Fixes #124

### CODE REVIEW

File: server/internal/middleware/hooks/gen_cros_hook.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+	"github.com/gin-gonic/gin"
+
+	urltrie "github.com/OpenIMSDK/OpenKF/server/internal/middleware/hooks/url_trie"
+)
+
+var _ urltrie.Hook = (*CROS)(nil)
+
+func init() {
+	urltrie.RegisterHook(&CROS{})
```
Comment (cubxxw) [inline server/internal/middleware/hooks/gen_cros_hook.go:29]: CROS???
Is that a misspelling ?

File: server/internal/middleware/hooks/gen_cros_hook.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+// Priority EDIT THIS TO YOUR OWN HOOK PRIORITY.
+func (h *CROS) Priority() int64 {
+	return 0
+}
+
+// BeforeRun EDIT THIS TO YOUR OWN HOOK BEFORE RUN, DO NOT NEED USE Next() FUNCTION.
+func (h *CROS) BeforeRun(c *gin.Context) {
+	// c.Writer.Header().Set("Access-Control-Allow-Origin", "*")
+	c.Writer.Header().Set("Access-Control-Allow-Origin", c.GetHeader("origin"))
```
Comment (cubxxw) [inline server/internal/middleware/hooks/gen_cros_hook.go:53]:
```suggestion
	c.Writer.Header().Set("Access-Control-Allow-Origin", c.GetHeader("origin"))
```
Whether to hardcode?

File: server/internal/middleware/hooks/gen_cros_hook.go
Comment (cubxxw) [inline server/internal/middleware/hooks/gen_cros_hook.go:1]: Consider adding error handling

File: server/internal/middleware/hooks/gen_jwt_hook.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	fmt.Println("RegisterHook", "Register Hook[JWT] success...")
+}
+
+// JWT implement urltrie.Hook.
+type JWT struct {
+	urltrie.Hook
+}
+
+// Patterns EDIT THIS TO YOUR OWN HOOK PATTERN.
+func (h *JWT) Patterns() []string {
```
Comment (cubxxw) [inline server/internal/middleware/hooks/gen_jwt_hook.go:41]: may can u use `patterns` ?

**Review by cubxxw — APPROVED** [review]

File: server/internal/middleware/hooks/gen_cros_hook.go (hunk 1 above)
Comment (AllianceTing) [reply server/internal/middleware/hooks/gen_cros_hook.go:29]: cors means cross origin？

**Review by AllianceTing — COMMENTED** [review]
please to solve those probs above

File: server/internal/middleware/hooks/gen_jwt_hook.go (hunk 3 above)
Comment (IRONICBo) [reply server/internal/middleware/hooks/gen_jwt_hook.go:41]: It's designed to public function.

File: server/internal/middleware/hooks/gen_cros_hook.go (hunk 2 above)
Comment (IRONICBo) [reply server/internal/middleware/hooks/gen_cros_hook.go:53]: These hard code are from http request headers, and they may not be used outside.

**Review by cubxxw — APPROVED** [review]

**Pull request merged**